addons/payment/controllers/portal.py

```diff
@@ -271,7 +273,7 @@ def payment_token(self, pm_id, reference, amount, currency_id, return_url=None,
     def confirm(self, **kw):
         tx_id = int(kw.get('tx_id', 0))
         if tx_id:
-            tx = request.env['payment.transaction'].browse(tx_id)
+            tx = request.env['payment.transaction'].sudo().browse(tx_id)
```

Comment:
why this changes?

Reply:
@dbo-odoo , when the payment is processed through the public user by direct link at that time it will not able to browse the transaction and due to that user was redirected to /my/home page instead of the transaction confirmation page.